File: lara1/app/Models/UserModel.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

use Illuminate\Support\Facades\DB;

class UserModel extends Model
{
    use HasFactory;

    protected $table = 'my_info';

    public function checkLogin($userName, $passwd) {

        $newPasswd = md5($passwd);

        $data = DB::table($this->table)->select(['uid', 'userName', 'phone'])->where(['userName'=>$userName, 'passwd'=>$newPasswd ])->first();

        if ($data) {
            $ret = ['result'=>SUCCESS, 'data'=>$data];
        } else {
            $ret = ['result'=> FAILED];
        }
        return $ret;

    }

    public function get_userInfo($uid) {
        if (! $uid) {
            $ret = ['result'=> FAILED, 'msg'=>'用户,未传入'];
            return $ret;
        }

        $data = [];
        $msg = '';
    
        $data = DB::table($this->table)->select(['uid', 'userName', 'phone'])->where(['uid'=>$uid ])->first();

        if ($data) {
            $ret = ['result'=>SUCCESS, 'data'=>$data];
        } else {
            $ret = ['result'=> FAILED];
        }
        return $ret;

    }

}

File: flask1/Model/UserModel.py
from dbExts import db

import hashlib

from flask import current_app as app


class UserModel(db.Model):
    __tablename__ = "my_info"

    uid = db.Column(db.Integer, primary_key=True)
    userName = db.Column(db.String(80))
    phone = db.Column(db.String(120))
    # addr = db.Column(db.addr)
    passwd = db.Column(db.Integer)


    # 登录
    def checkLogin(userName, passwd):
        
        md5_hash = hashlib.md5()
        md5_hash.update(passwd.encode('utf-8'))

        passwd_md5 = md5_hash.hexdigest()
        # 查询
        user = UserModel.query.filter_by(userName=userName, passwd=passwd_md5 ).with_entities(UserModel.userName, UserModel.uid, UserModel.phone ).first()

        # print(type(user))

        ret = {}
        if user:
            ret['result'] = app.config['SUCCESS']
            ret['data'] = user._asdict()
        else:
            ret['result'] = app.config['FAILED']
            ret['msg'] = '不存在'
        return ret;
    
    # 获取,用户信息
    def get_userInfo(uid):

        # 查询
        user = UserModel.query.filter_by(uid=uid ).with_entities(UserModel.userName, UserModel.uid, UserModel.phone ).first()

        # print(type(user))

        ret = {}
        if user:
            ret['result'] = app.config['SUCCESS']
            ret['data'] = user._asdict()
        else:
            ret['result'] = app.config['FAILED']
            ret['msg'] = '不存在'
        return ret;
    
